src/frontend/src/optimizer/plan_node/logical_over_window.rs

```diff
@@ -53,7 +53,41 @@ impl LogicalOverWindow {
         input: PlanRef,
         mut select_exprs: Vec<ExprImpl>,
     ) -> Result<(PlanRef, Vec<ExprImpl>)> {
-        let input_len = input.schema().len();
+        let mut input_proj_builder = ProjectBuilder::default();
+        let _: Vec<_> = input
+            .schema()
+            .fields()
+            .iter()
+            .enumerate()
+            .map(|(idx, x)| input_proj_builder.add_expr(&InputRef::new(idx, x.data_type()).into()))
+            .try_collect()
+            .map_err(|err| ErrorCode::NotImplemented(format!("{err} inside input"), None.into()))?;
+        let mut input_len = input.schema().len();
+        for expr in &select_exprs {
+            if let ExprImpl::WindowFunction(window_function) = expr {
+                let input_idx_in_order_by: Vec<_> = window_function
+                    .order_by
+                    .sort_exprs
+                    .iter()
+                    .map(|x| input_proj_builder.add_expr(&x.expr))
+                    .try_collect()
+                    .map_err(|err| {
+                        ErrorCode::NotImplemented(format!("{err} inside order_by"), None.into())
+                    })?;
+                let input_idx_in_partition_by: Vec<_> = window_function
+                    .partition_by
+                    .iter()
+                    .map(|x| input_proj_builder.add_expr(x))
+                    .try_collect()
+                    .map_err(|err| {
+                        ErrorCode::NotImplemented(format!("{err} inside partition_by"), None.into())
+                    })?;
```

Comment:
You might want to do this in `convert_window_function` since we currently have `partition_by` and `order_by` in `WindowFunction` and `PlanWindowFunction` structs instead of `OverWindow`. `ProjectBuilder` will do deduplication.

The conversion may be like:

```rs
fn convert_window_function(
    window_function: WindowFunction,
    proj_builder: &mut ProjectBuilder,
) -> Result<PlanWindowFunction> {
    let order_by: Vec<_> = window_function
        .order_by
        .sort_exprs
        .into_iter()
        .map(|e| {
            Ok(ColumnOrder::new(proj_builder.add_expr(&e.expr)?, e.order_type))
        })
        .try_collect()?;
    // ...
```

`PARTITION BY` and function arguments are similar.